test/core/credentials/transport/tls/spiffe_utils_test.cc

```diff
@@ -255,6 +294,208 @@ TEST(SpiffeId, TripleDotsSuccess) {
+          .status(),
+      absl::InvalidArgumentError(
+          "errors validating JSON: [field:trust_domains error:map key '' is "
+          "not a valid trust domain. INVALID_ARGUMENT: Trust domain cannot be "
+          "empty]"));
+}
+
+TEST(SpiffeBundle, InvalidTrustDomainFails) {
+  EXPECT_EQ(
+      SpiffeBundleMap::FromFile(
+          "test/core/credentials/transport/tls/test_data/spiffe/test_bundles/"
+          "spiffebundle_invalid_trustdomain.json")
+          .status(),
+      absl::InvalidArgumentError(
+          "errors validating JSON: [field:trust_domains error:map key "
+          "'invalid#character' is not a valid trust domain. INVALID_ARGUMENT: "
+          "Trust domain contains invalid character '#'. MUST contain only "
+          "lowercase letters, numbers, dots, dashes, and underscores]"));
+}
+
+TEST(SpiffeBundle, MalformedJsonFails) {
+  EXPECT_EQ(
+      SpiffeBundleMap::FromFile(
+          "test/core/credentials/transport/tls/test_data/spiffe/test_bundles/"
+          "spiffebundle_malformed.json")
+          .status(),
+      absl::InvalidArgumentError(
+          "errors validating JSON: [field: error:is not an object]"));
+}
+
+TEST(SpiffeBundle, WrongKtyFails) {
+  EXPECT_EQ(
+      SpiffeBundleMap::FromFile(
+          "test/core/credentials/transport/tls/test_data/spiffe/test_bundles/"
+          "spiffebundle_wrong_kty.json")
+          .status(),
+      absl::InvalidArgumentError(
+          "errors validating JSON: "
+          "[field:trust_domains[\"example.com\"].keys[0].kty "
+          "error:got EC. Only supported value for kty field is RSA]"));
+}
+
+TEST(SpiffeBundle, WrongKidFails) {
+  EXPECT_EQ(
+      SpiffeBundleMap::FromFile(
+          "test/core/credentials/transport/tls/test_data/spiffe/test_bundles/"
+          "spiffebundle_wrong_kid.json")
+          .status(),
+      absl::InvalidArgumentError(
+          "errors validating JSON: "
+          "[field:trust_domains[\"example.com\"].keys[0].kty errors:[field not "
+          "present; got . Only supported value for kty field is RSA]]"));
+}
+
+TEST(SpiffeBundle, MultiCertsFails) {
+  EXPECT_EQ(
+      SpiffeBundleMap::FromFile(
+          "test/core/credentials/transport/tls/test_data/spiffe/test_bundles/"
+          "spiffebundle_wrong_multi_certs.json")
+          .status(),
+      absl::InvalidArgumentError(
+          "errors validating JSON: "
+          "[field:trust_domains[\"google.com\"].keys[0].x5c "
+          "error:got vector length 2. Expected length of exactly 1.]"));
+}
+
+TEST(SpiffeBundle, WrongRootFails) {
+  EXPECT_EQ(
+      SpiffeBundleMap::FromFile(
+          "test/core/credentials/transport/tls/test_data/spiffe/test_bundles/"
+          "spiffebundle_wrong_root.json")
+          .status(),
+      absl::InvalidArgumentError("errors validating JSON: [field:trust_domains "
+                                 "error:field not present]"));
+}
+
+TEST(SpiffeBundle, WrongUseFails) {
```

Comment:
Do we have a test where SpiffeBundleMap::FromFile fails because one key is bad while the other key is good?

Reply:
When you say key do you mean a trust domain or something about one of the `keys` in the JSON `keys` list?
Added both
1. Added a fully valid trust domain in the failing trust domain bundle.
2. In `wrong_kty` I added a second key that is valid.

Reply:
I was thinking the latter, but both are good to have. Thanks!